docs/API_DESIGN.md

```diff
@@ -0,0 +1,652 @@
+resolution_result = DidJwk.resolve(uri)
+```
+
+### `DidWeb`
+
+```pseudocode!
+CLASS DidWeb
+  CONSTRUCTOR(uri: string)
+  STATIC METHOD resolve(uri: string): ResolutionResult
+```
+
+#### Example: Instantiate an existing `did:web`
+
+```pseudocode!
+uri = "did:web:example.com"
+did_web = new DidWeb(uri)
+```
+
+#### Example: Resolve a `did:web`
+
+```pseudocode!
+uri = "did:web:example.com"
+resolution_result = DidWeb.resolve(uri)
+```
+
+### `DidDht`
+
+```pseudocode!
+CLASS DidDht
+  PUBLIC DATA did: Did
+  PUBLIC DATA document: Document
+  CONSTRUCTOR(identity_key: Jwk)
+  CONSTRUCTOR(uri: string)
+  METHOD publish(signer: DsaSigner)
+  METHOD deactivate(signer: DsaSigner)
+  STATIC METHOD resolve(uri: string)
+```
+
+> [!NOTE]
+> `resolve()` makes use of [`Ed25519Verifier`](#ed25519verifier) internally for DNS packet verification.
+
+#### Example: Create & publish a `did:dht`
+
+```pseudocode!
+key_manager = new InMemoryKeyManager()
+public_key = key_manager.generate_key_material()
+did_dht = new DidDht(public_key)
+signer = key_manager.get_signer(public_key)
+did_dht.publish(signer)
+```
+
+#### Example: Create a `did:dht`, add to the Core Properties & publish
+
+> [!NOTE]
+> The call to the `new DidDht()` constructor only adds the minimum requirements to the DID Document.
+> If additional [Core Properties](https://www.w3.org/TR/did-core/#core-properties) are required, update the `document` data member prior-to the call to `publish()`.
+
+```pseudocode!
+key_manager = new InMemoryKeyManager()
+public_key = key_manager.generate_key_material()
+did_dht = new DidDht(public_key)
+
+/// Set the alsoKnownAs
+did_dht.document.alsoKnownAs = "did:example:efgh"
+/// Note: you could also add a verification method, set the controller etc.
+
+signer = key_manager.get_signer(public_key)
+did_dht.publish(signer)
+```
+
+#### Example: Instantiate an existing `did:dht`
+
+```pseudocode!
+uri = "did:dht:i9xkp8ddcbcg8jwq54ox699wuzxyifsqx4jru45zodqu453ksz6y"
+did_dht = new DidDht(uri)
+```
+
+#### Example: Update a `did:dht`
+
+```pseudocode!
+uri = "did:dht:i9xkp8ddcbcg8jwq54ox699wuzxyifsqx4jru45zodqu453ksz6y"
+did_dht = new DidDht(uri)
+
+/// Set the alsoKnownAs
+did_dht.document.alsoKnownAs = "did:example:efgh"
+/// Note: you could also add a verification method, set the controller etc.
+
+key_manager = new InMemoryKeyManager()
+public_key = key_manager.import_key(private_key) /// assume private_key pre-exists, eg. read from env var
+signer = key_manager.get_signer(public_key)
+
+did_dht.publish(signer)
+```
+
+#### Example: Resolve a `did:dht`
+
+```pseudocode!
+uri = "did:dht:i9xkp8ddcbcg8jwq54ox699wuzxyifsqx4jru45zodqu453ksz6y"
+resolution_result = DidDht.resolve(uri)
+```
+
+# Verifiable Credential's (VCs)
+
+## `NamedIssuer`
+
+```pseudocode!
+CLASS NamedIssuer
+  PUBLIC DATA id: string
+  PUBLIC DATA name: string
+```
+
+## `VerifiableCredential`
```

Comment:
it may make sense to namespace this, since we'll soon need to account for differnet types of 'verifiable credentials' or the more generic 'digital credentials' - for example: SD-JWT, SD-JWT-VC, mDL/mDoc, W3C VCDM v1 (what we currently support), W3C VCDM v2

Reply:
Moved under `Data Model 1.1` markdown section